### PR TITLE
[Build System] Refactor build-script.py and update-toolchain.py to use more idiomatic Python.

### PR DESCRIPTION
**Pull request by Rostepher**
This PR re-factors the `build-script.py` and `update-toolchain.py` scripts to use more idiomatic Python and conform to the standard PEP8 style guide. There are no functional changes, merely a re-organization of code and addition of comments. All functionality remains unchanged, just with improvements to control flow and output organization.

### CODE REVIEW

File: build-script.py [hunk 1]
```diff
@@ -1,498 +1,692 @@
...
+
+def note(*objects):
+    """Prints a note to stderr and then flushes the output.
+    """
+
+    _print('NOTE:', *objects, file=sys.stdout)
+
+
+def error(*objects):
+    """Prints an error message to stderr and then flushes the output.
```
Comment (harlanhaskins) [inline build-script.py]: Should these wrappers pass `flush=True`?

File: build-script.py [hunk 2]
```diff
@@ -1,498 +1,692 @@
...
+            GYB_EXEC,
+            os.path.join(sources_dir, file),
+            '-o', os.path.join(temp_dir, output_file_name),
+        ]
+
+        # Source locations are added by default by gyb, and cleared by passing
+        # `--line-directive=` (nothing following the `=`) to the generator. Our
+        # flag is the reverse; we don't want them by default, only if
+        # requested.
+        if add_source_locations:
```
Comment (harlanhaskins) [inline build-script.py]: Should this flag be negated?

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by Rostepher** [comment]
@swift-ci please test

**Comment by Rostepher** [comment]
@swift-ci please test

**Comment by Rostepher** [comment]
Alright I think I fixed the test failure, it was a small typo.

**Comment by Rostepher** [comment]
@swift-ci please test

**Comment by Rostepher** [comment]
@swift-ci please test

**Comment by Rostepher** [comment]
Closing stale PR. Might try to update this script again sometime in the future.

**Pull request closed without merging**